File: frontend/types/Course.ts
// Course interface matching Faculty_System database schema
export interface Course {
  Crs_ID: number;
  Crs_Name: string;
  Discription: string;
  Credit_Hours: number;
  Doc_ID: number;
  Dept_ID: number;
  Max_Num_Stu?: number;
  // Related data
  schedules?: CourseSchedule[];
  instructor?: {
    Doc_ID: number;
    FName: string;
    LName: string;
    Email: string;
  };
  department?: {
    Dept_ID: number;
    Dept_Name: string;
    Dept_Code?: string;
  };
}

export interface CreateCourseRequest {
  Crs_Name: string;
  Discription: string;
  Credit_Hours: number;
  Doc_ID: number;
  Dept_ID: number;
  Max_Num_Stu?: number;
}

export interface UpdateCourseRequest extends Partial<CreateCourseRequest> {}

export interface CourseSchedule {
  Sch_ID: number;
  Crs_ID: number;
  Day: 'Saturday' | 'Sunday' | 'Monday' | 'Tuesday' | 'Wednesday' | 'Thursday';
  Start_Hour: string; // time format
  END_Hour: string; // time format
  Level: 1 | 2 | 3 | 4;
  Location?: string;
}

export interface Enrollment {
  Stu_ID: number;
  Crs_ID: number;
  Grade?: number; // 0-100
  Year: string; // e.g., '2024-25'
}

export interface Teach {
  Crs_ID: number;
  Doc_ID: number;
  Semester: string;
}

export interface CourseWithSchedule extends Course {
  schedules: CourseSchedule[];
}

File: api/Models/Course.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    [Table("Course", Schema = "Faculty")]
    public class Course
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Crs_ID { get; set; }

        [Required]
        [StringLength(50)]
        public string Crs_Name { get; set; } = string.Empty;

        [Required]
        [StringLength(500)]
        public string Discription { get; set; } = string.Empty;

        [Required]
        public int Credit_Hours { get; set; }

        [Required]
        public int Doc_ID { get; set; }

        [Required]
        public int Dept_ID { get; set; }

        public int? Max_Num_Stu { get; set; }

        // Navigation properties
        [ForeignKey("Doc_ID")]
        public virtual Doctor? Instructor { get; set; }

        [ForeignKey("Dept_ID")]
        public virtual Department? Department { get; set; }

        public virtual ICollection<CourseSchedule> Schedules { get; set; } = new List<CourseSchedule>();
        public virtual ICollection<Enrollement> Enrollments { get; set; } = new List<Enrollement>();
        public virtual ICollection<Teach> TeachingAssignments { get; set; } = new List<Teach>();
    }
}

